crates/forge/tests/cli/revive_compiler.rs

```diff
@@ -0,0 +1,298 @@
+        config.resolc.resolc = Some(foundry_config::SolcReq::Version(
+            semver::Version::parse(OTHER_RESOLC_VERSION).unwrap(),
+        ));
+        config.resolc.resolc_compile = true;
+    });
+
+    prj.add_source("ContractA", CONTRACT_A).unwrap();
+
+    cmd.args(["compiler", "resolve", "--root", prj.root().to_str().unwrap()])
+        .assert_success()
+        .stdout_eq(str![[r#"
+Solidity:
+- Resolc v0.1.0-dev.13 and Solc v0.8.4
+
+
+"#]]);
+});
+
+forgetest!(can_list_resolved_compiler_versions, |prj, cmd| {
+    prj.update_config(|config| {
+        config.resolc.resolc = Some(foundry_config::SolcReq::Version(
+            semver::Version::parse(OTHER_RESOLC_VERSION).unwrap(),
+        ));
+        config.resolc.resolc_compile = true;
+    });
+    prj.add_source("ContractA", CONTRACT_A).unwrap();
+
+    cmd.args(["compiler", "resolve"]).assert_success().stdout_eq(str![[r#"
+Solidity:
+- Resolc v0.1.0-dev.13 and Solc v0.8.4
+
+
+"#]]);
+});
+
+forgetest!(can_list_resolved_compiler_versions_json, |prj, cmd| {
+    prj.update_config(|config| {
+        config.resolc.resolc = Some(foundry_config::SolcReq::Version(
+            semver::Version::parse(OTHER_RESOLC_VERSION).unwrap(),
+        ));
+        config.resolc.resolc_compile = true;
+    });
+    prj.add_source("ContractA", CONTRACT_A).unwrap();
+
+    cmd.args(["compiler", "resolve", "--json"]).assert_success().stdout_eq(
+        str![[r#"
+{
+  "Solidity": [
+    {
+      "name": "Resolc",
+      "version": "0.1.0-dev.13",
+      "dep": [
+        "Solc",
+        "0.8.4"
+      ]
+    }
+  ]
+}
+"#]]
+        .is_json(),
+    );
+});
+
+forgetest!(can_list_resolved_compiler_versions_verbose, |prj, cmd| {
+    prj.update_config(|config| {
+        config.resolc.resolc = Some(foundry_config::SolcReq::Version(
+            semver::Version::parse(OTHER_RESOLC_VERSION).unwrap(),
+        ));
+        config.resolc.resolc_compile = true;
+    });
+    prj.add_source("ContractC", CONTRACT_C).unwrap();
+    prj.add_source("ContractD", CONTRACT_D).unwrap();
+
+    cmd.args(["compiler", "resolve", "-v"]).assert_success().stdout_eq(str![[r#"
+Solidity:
+
+Resolc v0.1.0-dev.13 and Solc v0.8.27:
+├── src/ContractC.sol
+└── src/ContractD.sol
+
+
+"#]]);
+});
+
+forgetest!(can_list_resolved_compiler_versions_verbose_json, |prj, cmd| {
+    prj.update_config(|config| {
+        config.resolc.resolc = Some(foundry_config::SolcReq::Version(
+            semver::Version::parse(OTHER_RESOLC_VERSION).unwrap(),
+        ));
+        config.resolc.resolc_compile = true;
+    });
+    prj.add_source("ContractC", CONTRACT_C).unwrap();
+    prj.add_source("ContractD", CONTRACT_D).unwrap();
+
+    cmd.args(["compiler", "resolve", "--json", "-v"]).assert_success().stdout_eq(
+        str![[r#"
+{
+  "Solidity": [
+    {
+      "name": "Resolc",
+      "version": "0.1.0-dev.13",
+      "paths": [
+        "src/ContractC.sol",
+        "src/ContractD.sol"
+      ],
+      "dep": [
```

Comment:
why is it array - can we have more than one solc dep?
Maybe?
```
"dependency":
 {
    "name": "Solc",
    "version": "0.8.27"
  }
 ```

Reply:
it's an array because tuples get serialized as such, moved the fields to be a struct